app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,56 @@
+class CommentsController < ApplicationController
+  before_action :find_post
+  before_action :find_comment, only: [:edit, :update, :destroy]
+
+  def create
+    @comment = @post.comments.new comment_params
+    if @comment.save
+      respond_to do |format|  
+        format.html { redirect_to @comment.post, notice: 'Comment was successfully created.' }
+        format.js { render layout: false }
+        format.json { render @comment.post, status: :created, location: @comment }
+      end
+    else 
+      respond_to do |format|  
+        format.html { redirect_to @comment.post, notice: 'Please sign or sign up to comment' }
+      end
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @comment.update(comment_params)
+      flash[:notice] = 'Update successfully'
+    else
+      flash[:danger] = 'Update unsuccessfully'
+    end
+
+    redirect_to post_path(@post.id)
+  end
+
+  def destroy
+    if @comment.destroy
+      respond_to do |format|  
+        format.html { redirect_to @comment.post, notice: 'Delete successfully' }
+        format.js { render layout: false }
+        format.json { render @comment.post, status: :created, location: @comment }
+      end
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:content, :user_id, :post_id)
```

Comment:
Don't use `user_id` as `hidden field` to submit params like this. you can user `@coment.user = current_user`. It will be better, to prevent someone use their account but submit comment for another user - they can inspect and value of `user_id` in your form.

`post_id` shouldn't use in your form in this case. you can get `post_id` from the route